index.js

```diff
@@ -46,15 +44,55 @@ const register = function(server, options) {
     timing.elapsed = timing.end - timing.start;
   });
 
-  server.events.on('response', (request) => {
-    // check the tail response times and notify if needed:
-    const responseTime = request.info.responded - request.info.received;
-    const plugin = request.route.settings.plugins['hapi-timing'];
-    const threshold = (plugin && plugin.threshold) ? plugin.threshold : options.threshold;
-    if (responseTime > threshold) {
-      requestTimeoutExpired(responseTime, threshold, request);
-    }
-  });
+  if (options.requestLifecycle) {
+    const eventNames = ['onRequest', 'onPreAuth', 'onPostAuth', 'onPreHandler', 'onPostHandler'];
+    eventNames.forEach(eventName => {
```

Comment:
I think we should use the timerStart and end and just have the next event call timerEnd for the one before